public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,188 @@
+        repo='https://github.com/iterative/dataset-registry'
+        ) as fd:
+    # ... fd is a file descriptor that can be processed normally.
+```
+
+## Description
+
+Open file or model (`path`) tracked in a <abbr>DVC project</abbr> (by DVC or
+Git), and generate a corresponding
+[file object](https://docs.python.org/3/glossary.html#term-file-object).
+
+> The exact type of file object generated depends on the `mode` used. For more
+> details, please refer to Python's
+> [`open()`](https://docs.python.org/3/library/functions.html#open) built-in,
+> which is used under the hood.
+
+`dvc.api.open()` may only be used as a
+[context manager](https://www.python.org/dev/peps/pep-0343/#context-managers-in-the-standard-library)
+(using the `with` keyword, as shown in the **Examples** below).
+
+> Use `dvc.api.read()` to get the file's contents directly – no _context
+> manager_ involved.
+
+This function reads (streams) the file trough a direct connection to the storage
+whenever possible, so it does not require any space on the disc to save the file
+before making it accessible. The only exception is when using a Google Drive
+[remote type](/doc/command-reference/remote/add#supported-storage-types).
+
+## Parameters
+
+- **`path`** - location and file name of the file in `repo`, relative to the
+  project's root.
+
+- `repo` - specifies the location of the DVC project. It can be a URL or a file
+  system path. Both HTTP and SSH protocols are supported for online Git repos
+  (e.g. `[user@]server:project.git`). _Default_: The current project is used
+  (the current working directory tree is walked up to find it).
+
+- `rev` - Git commit (any [revision](https://git-scm.com/docs/revisions) such as
+  a branch or tag name, or a commit hash). If `repo` is not a Git repo, this
+  option is ignored. _Default_: `HEAD`.
+
+- `remote` - name of the [DVC remote](/doc/command-reference/remote) to look for
+  the target data. _Default_: The
+  [default remote](/doc/command-reference/remote/default) of `repo` is used if a
+  `remote` argument is not given. For local projects, the <abbr>cache</abbr> is
+  tied before the default remote.
+
+- `mode` - specifies the mode in which the file is opened. Defaults to `"r"`
+  (read). Mirrors the namesake parameter in builtin
+  [`open()`](https://docs.python.org/3/library/functions.html#open).
+
+- `encoding` -
+  [codec](https://docs.python.org/3/library/codecs.html#standard-encodings) used
+  to decode the file contents to a string. This should only be used in text
+  mode. Defaults to `"utf-8"`. Mirrors the namesake parameter in builtin
+  `open()`.
+
+## Exceptions
+
+- `dvc.exceptions.FileMissingError` - file in `path` is missing from `repo`.
+
+- `dvc.exceptions.PathMissingError` - `path` cannot be found in `repo`.
+
+- `dvc.api.UrlNotDvcRepoError` - `repo` is not a DVC project.
+
+- `dvc.exceptions.NoRemoteError` - no `remote` is found.
+
+## Example: Use data tracked in a DVC repository online
```

Comment:
I don't quite understand the name, to be honest

Reply:
How about "Use data from a DVC repository online"?

Reply:
I would remove `online` ... instead, in rare cases we can figure out how to specify "offline"/local repos ... if it's needed at all

Reply:
I see the word "online" is throwing you off. I removed it from the text below but I'm not sure how best to replace it here. I can't say "remote DVC repo" because that's confusing: DVC remotes are not repositories but storage (cache backups, basically).
"...repo on the cloud" ? Sounds like marketing.
"...repo on Github" ? Too specific for the title.
Notice the word "online" is also used in the explanation of the `repo` param, so maybe it's OK to keep it here too.

Reply:
Just omit it? :)

Reply:
OK. I guess using local repos is an edge case? Currently that's only shown as a side note in the 2nd example (and explained in the `remote` param). Removed.